### DISCUSSION TITLE
Hyperhdr v19 no light despite running as root

### DISCUSSION THREAD

**Discussion by KnoppKnopp** [awawa-dev/HyperHDR#629] [Q&A]
Hi all,

I'm racking my brains trying to get these to work.

I have installed v19.0.0.0 onto my pi 3b and enabled root user but I still cannot get my ws2812b lights on through gpio18 using ws281x controller.

I have verified it is running as root here:

● hyperhdr@root.service - HyperHdr ambient light systemd service  for user root
     Loaded: loaded (/etc/systemd/system/hyperhdr@.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-08-25 11:05:27 BST; 7min ago
   Main PID: 509 (hyperhdr)
      Tasks: 11 (limit: 1595)
        CPU: 1.916s
     CGroup: /system.slice/system-hyperhdr.slice/hyperhdr@root.service
             └─509 /usr/bin/hyperhdr

The log from the hyperhdr ui seemingly is normal.

One more point to add is I have connect the hyperion mobile controller app to my rasberry pi and I can see the first led in the strip lighting up when I change the colour but it does not correspond to the colour I have selected, its just a very faint light which stops when I stop selecting a colour.

I would also like to add when checking the running users services i get this:

jack@raspberrypi:~ $ ps aux | grep hyperhdr
root       509  0.1  3.2 184772 30748 ?        Ssl  11:05   0:02 /usr/bin/hyperhdr
jack       744  0.0  0.0   7452   508 pts/0    S+   11:37   0:00 grep --color=auto hyperhdr

Any help would be appreciated.

**Comment by awawa-dev** [comment]
Hi
I usually do not diagnose problems without full logs and using not supported led drivers including missing 3.3v to 5v level shifter. If there is no errors in the log then the problem is probably hardware, too long cables/interference or has electric nature (e.g. missing common ground if there are separate power supplies, bad cabling). You can continue to waste time diagnosing such a setup, which is impossible task without the right equipment (oscilloscope or logic analyzer with customizable voltage triggering level) or at least start by adding a proper 3.3v to 5v level shifter to your current system. Or save a lot of time by investing ~$10 in recommended models of Pico board that has an integrated level shifter. The choice is yours.

**Reply by KnoppKnopp** [reply]
Thanks for the reply.

I have a 3.3v to 5v shifter. I'll try it.
I'll also look into the Pico board. I hadn't realised they existed.
I followed a youtube tutorial as did many probably so apologies if it sounded like an ill informed request.

**Reply by awawa-dev** [reply]
Pico is a new solution, so you may not have heard of it yet. A lot has changed recently. We now have integrated solutions. If we don't want to deal with building circuits with a level shifter, we can buy a cheap ready-made solution e.g. Adafruit ItsyBitsy/Scorpio or Pimoroni Plasma/Stick. Their availability is very good at least when it comes to US, UK and EU, I have no idea about other markets.

**Reply by KnoppKnopp** [reply]
I see. I will definitely look into that.

I have just tried the logic level converter and it has somewhat worked.
The lights flicker when the brightness falls below ~40% (ie. a black screen). I'm assuming this is something to do with the common grounds. I'll experiment.

**Reply by awawa-dev** [reply]
I do not advise you to blindly experiment when connecting something via GPIO, because you risk your Raspberry Pi ;)
If you use separate power supplies, a common ground between the LED strip and the Rpi is simply mandatory. Not only can the LED not work without it, but there is also a small chance of damage to the peripherals. Another explanation could be a signal problem: the RPi is simply not designed for such tasks of emulating Neopixel protocol, or you are using the wrong type of level shifter (too slow, I2C, neopixel low may be randomly promoted to high, so it flickers at low brightness) or too much noise EMI/cable too long.

**Reply by KnoppKnopp** [reply]
I see.

Thank you for you advice.
I am going to try one of those pico boards.

Will this work alongside the pi to drive the LEDs depending on the video source? or will it be independently driving the LEDs and receiving the video source?

I have found the hyperserialpico page so I can research myself to save you the hassle!

**Reply by KnoppKnopp** [reply]
Hi,

Sorry to bother you about this again.

I have my Adafruit Scorpio and i'm following through the guide: https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html#chapter1

It mentions HyperHDR V.20 is needed but I cannot find a V.20 release. Apologies if it is something that is obvious but I cannot see a v20, only v19 which I have installed.

I have flashed my Scorpio with the ws2812 version of the Scorpio .uf2, found here: https://github.com/awawa-dev/HyperSerialPico/releases

Best.

**Reply by awawa-dev** [reply]
Hi
Use installers from latest build of the master branch on Github Action. Curretnly https://github.com/awawa-dev/HyperHDR/actions/runs/6017162553

**Reply by KnoppKnopp** [reply]
Hi I'm really sorry but im not sure how to install it from that page.

**Reply by awawa-dev** [reply]
Maybe this will help? https://github.com/awawa-dev/HyperHDR/discussions/621#discussioncomment-6704284

**Reply by KnoppKnopp** [reply]
I downloaded and extracted the deb file in windows then copied that deb file to my pi using scp and attempted to install it but it still will not work.

Sorry to be a bother but I'm very novice at this. Perhaps there is instructions for installing v20 from the archives?

I will keep trying.

**Reply by awawa-dev** [reply]
Are you sure you've installed the right version (32bit or aarch64 Bullseye)? And uninstalled the old one first? Just follow wiki: https://github.com/awawa-dev/HyperHDR/wiki/Installation#linux there is an information what command to run in case of problems (with -d) to see the details.

**Reply by KnoppKnopp** [reply]
Hi.

Yes I am sure. I understand how to install deb files, just the v20 beta is in a packaged folder so it will not direct download to my pi. That's why I tried to scp to over from my PC but for some reason wouldn't work.

Sorry to be a pain, this is likely something simple.

**Reply by KnoppKnopp** [reply]
Hi.

Apologies. after some fiddling, I have managed to instal the correct hyper deb onto my pi.
Now to get it talking to the Adafruit! Is there a method to verify the pi is recognising the Adafruit?

**Reply by awawa-dev** [reply]
Since you're also using Windows, HyperSerialPico should report in "Device Manager" under the "Port (COM and LPT)" tab. Try connecting/disconnecting Pico and you should see that tab refresh/change. I think this is the easiest way to check if the device is working properly at all. Then you can connect it to the Rpi. In the Adalight HyperHDR driver, you should find it listed (yellow button, you may need to select the "refresh" item first).

**Reply by KnoppKnopp** [reply]
That worked thank you and confirmed the Adafruit is working.
Although now it will not appear in HyperHDR.
Only the bluetooth option as mentioned in the quick start guide.

I'm so close i can feel it xD

![image](https://github.com/awawa-dev/HyperHDR/assets/92688938/be4cff74-d02e-4d45-9c87-e282797d95dd)

**Reply by awawa-dev** [reply]
This is Pico. The Bluetooth device is usually ttyAMA0 (but this can change) and has no name. Just test: unplug Pico and "Update" list: Pico/ttyACM0 should be gone. Reconnect, "Update" and it should show up again.

**Reply by KnoppKnopp** [reply]
That worked, so thats good. 

Now to figure out the wiring between the pi and the Adafruit. and the lights.
Is there a diagram for this anywhere?

Thank you for your help. If I can support the project anywhere I'd love to.

**Reply by awawa-dev** [reply]
If you are satisfied, just give the project a star is enough but it is not required ;) If you are building a classic system with one LED segment, you need two Scorpio pins for ws2812b https://learn.adafruit.com/introducing-feather-rp2040-scorpio/pinouts NEOPIXEL0 is used to transmit the data to the LED strip data channel, and its neighbor "GND" (ground). If you are using two power supplies (one for RPi/Scorpio and one for the LED strip), you must connect this Scorpio GND to the LED strip ground to ensure a common ground between the two power supplies. You can use the analogous schematic for esp32(HyperSerialESP32 version) from the last blog post, the Scorpio already has a level shifter built in so you don't need an external one (with the 5V output on pin NEOPIXEL0-7  it can only be connected to the LED, don't accidentally short it to another pin on the Scorpio or RPi)

**Reply by KnoppKnopp** [reply]
Thank you.

So the signal comes from the pi via the usb cable and the neopixel0 sends that data to the leds?

**Reply by awawa-dev** [reply]
Yes, Rpi -> USB cable -> USB power supply for Scorpio board, Data for the LED strip-> Scoprio-> shared GND with the LED strip ground and NEOPIXEL0 for the LED strip data channel

**Reply by KnoppKnopp** [reply]
Mate, it's worked flawlessly.

I massively appreciate your help. I appreciate it's tedious to help novice's out so helping me from my initial problem to now having it working is very kind of you.

**Reply by KnoppKnopp** [reply]
Final thing, just a small one!

Is there a way to default HDR to be toggled on everytime I start up hyperhdr?

Basicslly so I don't have to do it on the remote control tab every time.

**Reply by awawa-dev** [reply]
HyperHDR should start LEDs automatically if it has a video source or at least the background effect enabled. Check your logs (or post logs on pastebin.com and give the link) why HyperHDR decides to turn the LEDs off at the startup.

**Reply by KnoppKnopp** [reply]
Sorry.
I meant the HDR slider in remote control. Is there a way to default so it stays on everytime I start HyperHDR?

**Reply by awawa-dev** [reply]
You need to activate it in the grabber settings and save them.
